### PR TITLE
Groupedlist accessibility fixes

### PR DESCRIPTION
**Pull request by AbdouMoumen**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes
Accessibility fixes for GroupedList. Namely, ability to expand/collapse using arrow keys, and proper aria tree semantics for screen-readers.
Also, minor typo correction in Selection.ts

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by betrue-final-final** [comment]
I'd like to test this before merging.

File: packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.tsx [hunk 1]
```diff
@@ -73,7 +74,13 @@ export class GroupHeader extends BaseComponent<IGroupDividerProps, IGroupHeaderS
         })}
         style={viewport ? { minWidth: viewport.width } : {}}
         onClick={this._onHeaderClick}
+        onKeyDown={this._onKeyDown}
         aria-label={group.ariaLabel || group.name}
+        aria-selected={group.isSelected}
+        aria-expanded={!group.isCollapsed}
+        aria-level={(groupLevel || 0) + 1}
+        aria-posinset={(groupIndex || 0) + 1}
+        aria-setsize={group.children ? group.children.length : group.count}
```
Comment (aditima) [inline packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.tsx]: I would just go with group.count. group.children contains the sub-groups, if any, so a count of that is not an accurate reflection of number of items in that group. Also, we will possibly end up with something like ParentGroup (size=2), SubGroup1 (size = 10), SubGroup2 (size = 20) - that looks weird.

File: packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.tsx [hunk 2]
```diff
@@ -152,6 +159,18 @@ export class GroupHeader extends BaseComponent<IGroupDividerProps, IGroupHeaderS
     }
   };
 
+  private _onKeyDown = (evt: React.KeyboardEvent<HTMLElement>) => {
+    const { group, onToggleCollapse } = this.props;
+    const { isCollapsed } = group!;
+
+    if ((evt.keyCode === KeyCodes.right && isCollapsed) || (evt.keyCode === KeyCodes.left && !isCollapsed)) {
+      onToggleCollapse!(group!);
```
Comment (aditima) [inline packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.tsx]: Consider using the '!' operator only when absolutely necessary. For example, calling the onToggleCollapse method only makes sense if you have a valid group object.

if (onToggleCollapse && group) {
   onToggleCollapse(group);

   // we should probably continue bubbling the event if we did not handle it
   evt.stopPropagation();
   evt.preventDefault();
}

Comment (cliffkoh) [reply packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.tsx]: Usage of `!` should be so scarce that we should leave a comment justifying the usage.

**Comment by dzearing** [comment]
Just a friendly ping to @AbdouMoumen ; there are conflicts now. Please merge latest master into your branch. Once you fix those, auto deploy should provide a link which @betrue-final-final can use to play with your changes. If you want to give him some directions on what he should be looking at in the demo page to test your changes with edge/narrator, that would be helpful!

**Comment by cliffkoh** [comment]
Second friendly ping to @AbdouMoumen.

**Comment by dzearing** [comment]
As this has not received traction in more than 2 weeks, closing. Please re-open when you can address the feedback and resolve conflicts.

**Pull request closed without merging**